cloudwatchlogs_formatter.go

```diff
@@ -80,26 +95,93 @@ func (f *CloudWatchLogsFormatter) MsgCount() int {
 	return f.batch.MsgCount() + len(f.eData)
 }
 
-// Request to submit for the batch. See HTTPFormatter for more info.
+// Request constructs a request for this formatter
 func (f *CloudWatchLogsFormatter) Request() (*http.Request, error) {
-	body, err := newCloudWatchLogsBody(f.batch, f.eData, f.logGroupName, f.logStreamName, <-f.tokens)
+	if f.client == nil {
+		return nil, fmt.Errorf("CloudWatch Logs client is not initialized")
+	}
+
+	// Get the sequence token
+	token := <-f.tokens
+	if token == "" {
+		return nil, fmt.Errorf("invalid sequence token")
+	}
+
+	// Create the request body
+	body, err := newCloudWatchLogsBody(f.batch, f.eData, f.logGroupName, f.logStreamName, token)
 	if err != nil {
 		return nil, err
 	}
-	r, err := http.NewRequest(http.MethodPost, f.url, body)
+
+	// Create a new request
+	req, err := http.NewRequest(http.MethodPost, f.url, body)
```

Comment:
non-blocking -- should probably be NewRequestWithContext particularly if we have a ctx we can pass in here.

Reply:
I agree, though the `NewCloudWatchLogsFormatterFunc` doesn't receive a context. Happy to change the signature here if you think that's worth it.

Reply:
I ended up doing this for prior services because of a golangci-lint rule that required all NewRequest calls to be NewRequestWithContext - it's useful when/if instrumentation libraries are added in since all those do instrumentation via the Context.

I don't want that note to block this PR, though.  It's a separate issue.